luaui/Widgets/unit_immobile_builder.lua

```diff
@@ -132,10 +132,14 @@ end
 
 function widget:UnitCommand(unitID, unitDefID, _, cmdID, _, cmdOpts)
 	if isImmobileBuilder[unitDefID] and cmdOpts.shift and cmdID ~= CMD_FIGHT then
-		local commandQueue = spGetCommandQueue(unitID, -1)
-		local lastCommand = commandQueue[#commandQueue]
+		local lastCommand = spGetCommandQueue(unitID, 1)[1]
 		if lastCommand and lastCommand.id == CMD_FIGHT then
```

Comment:
`spGetCommandQueue(unitID, 1)[1]` gets the first command, not the last.

Reply:
the changes in this widget also dont look needed for this PR? or are they?

Reply:
It's hard to measure whether the new widget has acceptable performance, as `unit_immobile_builder` causes these actions to already have terrible performance.

Reply:
I understand but those changes should go at a different PR then improvements can be merged separately.

Reply:
here you go: https://github.com/beyond-all-reason/Beyond-All-Reason/pull/4018

Reply:
merged #4018